70 tripal job test

4ctrl-alt-del opened the pull request:
Implemented functional testing for the Tripal Job service. As a result fixed multiple deprecation warnings from the Tripal Job service class.

4ctrl-alt-del committed:
Added tripal job test.

Added initial PHP file and class that tests the tripal job service. Added first
initial test that verifies creation of jobs works.

Updated deprecated use of format date to new object oriented method in drupal 8.

4ctrl-alt-del committed:
Worked on tripal job test.

Added unit tests for loading tripal job. Added unit tests for all getter methods
of a tripal job class.

All unit tests pass.

4ctrl-alt-del committed:
Worked on tripal job test.

Added unit test for testing set progress.

Started working on unit testing the run method, running into much difficulty
with trying to get a thread going to test the state of the tripal job as it is
running.

4ctrl-alt-del committed:
Worked on tripal job test.

Added unit test for tripal job run method.

4ctrl-alt-del committed:
Worked on tripal job test.

Added unit tests for items handled methods and progress status. For some reason
the tripal job class was printing things to standard output in its set items
handled method which from what I can tell a drupal service implementation should
never do, therefore the print segments of code were commented out.

4ctrl-alt-del committed:
Worked on tripal job test.

Added comments to the unit test class.

spficklin commented on tripal/src/Services/TripalJob.php:
```diff
@@ -530,11 +531,11 @@ public function setItemsHandled($total_handled) {
-      print "Percent complete: 0%. Memory: " . $memory . " bytes.\r";
-      return;
-    }
+    //if ($total_handled == 0) {
```
We need to find a way to keep these print lines without failing the unit test.

4ctrl-alt-del committed:
Added interface.

Added tripal job interface, which in itself is not a drupal plugin because it
needs to be created in two different ways; creation and loading. Still figuring
out how to do that.

4ctrl-alt-del committed:
Added interface.

Figured out a way to have to methods for constructing tripal jobs; make a
tripal job factory plugin, which in turn creates or loads a tripal job.

4ctrl-alt-del committed:
Added plugin manager.

Added plugin manager and annotation class for the tripal job factory plugin.

4ctrl-alt-del committed:
Added executor plugin.

Added new plugin interface, plugin manager, and annotation class for a new tripal executor plugin.

4ctrl-alt-del committed:
Added to tripal job interface.

Added additional interface methods for the job factory interface and job interface classes.

laceysanderson commented:
Note: I'm going to close this PR as the original functionality for testing has shown us that we want to change the structure of jobs. @4ctrl-alt-del is currently working on this in the current branch but as it's not ready yet I'm going to close the PR. @4ctrl-alt-del when the functionality is ready please resubmit the PR! :-)
